digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/middleware/TrafficSignManager.scala

```diff
@@ -89,22 +87,26 @@ case class TrafficSignManager(manoeuvreService: ManoeuvreService, roadLinkServic
           manoeuvreService.deleteManoeuvreFromSign(manoeuvreService.withIds(Set(trSign.id)), username, newTransaction)
 
         case signType if TrafficSignManager.belongsToProhibition(signType) =>
-          insertTrafficSignToProcess(trSign.id, Prohibition, Some(trSign))
+          insertTrafficSignToProcess(trSign.id, Prohibition, Some(trSign), newTransaction)
 
         case signType if TrafficSignManager.belongsToHazmat(signType) =>
-          insertTrafficSignToProcess(trSign.id, HazmatTransportProhibition, Some(trSign))
+          insertTrafficSignToProcess(trSign.id, HazmatTransportProhibition, Some(trSign), newTransaction)
 
         case signType if TrafficSignManager.belongsToParking(signType) =>
-          insertTrafficSignToProcess(trSign.id, ParkingProhibition, Some(trSign))
+          insertTrafficSignToProcess(trSign.id, ParkingProhibition, Some(trSign), newTransaction)
 
         case _ => None
       }
     }
   }
 
-  def insertTrafficSignToProcess(id: Long, assetInfo: AssetTypeInfo, persistedTrafficSign: Option[PersistedTrafficSign] = None) : Unit = {
+  def insertTrafficSignToProcess(id: Long, assetInfo: AssetTypeInfo, persistedTrafficSign: Option[PersistedTrafficSign] = None, newTransaction: Boolean = false) : Unit = {
```

Comment:
this change could be reverted?

Reply:
Due the possibility of nasted transaction, we should keep this change.
If this function is called after TrafficSignTierekisteriImporter.expireAssets we will create a nasted transaction and with this change we can avoid the issue